### DISCUSSION TITLE
Use local tree id instead of global tree id in the geometry interface

### DISCUSSION THREAD

**Discussion by Davknapp** [DLR-AMR/t8code#896] [General]
We usually iterate over local elements, and request process local information. Apart from functions that are use to create a cmesh (set_vertices etc.) we should always use a local tree id in my opinion. Otherwise it results in multiple calls of ```t8_cmesh_get_global/local_id```. Is there another good reason to use global ids in function that are not used to create cmeshes?

**Comment by holke** [comment]
My initial line of thought was that the geometry is quite independent from any other information and that a caller may only have the global id available.

Looking at it from a different angle, your points are valid. Also 
- we need the cmesh to call the functions and we can always use the global_to_local computation should we only have the global id
- geometry information will probably be partitioned, so it does not make sense to call it for non-local trees


Maybe indices of ghost trees might be an issue...

I would suggest trying to change it to local and see if it breaks.

**Reply by Davknapp** [reply]
Will do that, thanks for sharing your thaughts on this. I will try it and keep you up to date ;)
